src/dotty/tools/dotc/core/TyperState.scala

```diff
@@ -59,6 +59,20 @@ class TyperState(r: Reporter) extends DotClass with Showable {
   /** Commit state so that it gets propagated to enclosing context */
   def commit()(implicit ctx: Context): Unit = unsupported("commit")
 
+  /** The typer state has already been committed */
+  def isCommitted: Boolean = false
+
+  /** Optionally, if this is a mutable typerstate, it's creator state */
+  def parent: Option[TyperState] = None
+
+  /** The closest ancestor of this typer state (including possible this typer state itself)
+   *  which is not yet committed.
```

Comment:
The documentation should state what is returned when this state and all of its ancestors are committed, or this method should return an `Option`